lib/mongoose_push/service/apns/supervisor.ex

```diff
@@ -0,0 +1,72 @@
+defmodule MongoosePush.Service.APNS.Supervisor do
+  @moduledoc """
+  APNS module supervising Sparrow's PoolSupervisor and APNS State
+  """
+  use Supervisor, id: :apns_supervisor
+  require Logger
+  alias MongoosePush.Application
+
+  @default_endpoints %{
+    dev: "api.development.push.apple.com",
+    prod: "api.push.apple.com"
+}
+
+  @spec start_link([Application.pool_definition()]) :: Supervisor.on_start()
+  def start_link(arg) do
+    Supervisor.start_link(__MODULE__, arg)
+  end
+
+  @impl true
+  def init(pool_configs) do
+    sparrow_config = create_sparrow_config(pool_configs)
+
+    children = [
+      Supervisor.child_spec({Sparrow.APNS.Pool.Supervisor, sparrow_config},
+        id: :apns_pool_supervisor
+      ),
+      {MongoosePush.Service.APNS.State, pool_configs}
+    ]
+
+    Supervisor.init(children, strategy: :one_for_one)
+  end
+
+  defp create_sparrow_config(pool_configs) do
+    {dev_list, prod_list} = List.foldl(pool_configs, {[], []}, &convert_pool_to_sparrow/2)
```

Comment:
Now that I think about it, I think it would be much simpler to read to split this into 2 steps:
1. We split the original list into 2: `prod` and `dev.`
2. Then on both lists, separately, we run `convert_pool_to_sparrow`. This way, we don't need this big and not very pleasant case at the end of `convert_pool_to_sparrow`.